core/src/main/java/kafka/log/remote/RemoteLogReader.java

```diff
@@ -31,14 +32,18 @@ public class RemoteLogReader implements Callable<Void> {
     private final Logger logger;
     private final RemoteStorageFetchInfo fetchInfo;
     private final RemoteLogManager rlm;
+    private final BrokerTopicStats brokerTopicStats;
     private final Consumer<RemoteLogReadResult> callback;
 
     public RemoteLogReader(RemoteStorageFetchInfo fetchInfo,
                            RemoteLogManager rlm,
-                           Consumer<RemoteLogReadResult> callback) {
+                           Consumer<RemoteLogReadResult> callback,
+                           BrokerTopicStats brokerTopicStats) {
         this.fetchInfo = fetchInfo;
         this.rlm = rlm;
+        this.brokerTopicStats = brokerTopicStats;
         this.callback = callback;
+        this.brokerTopicStats.topicStats(fetchInfo.topicPartition.topic()).remoteReadRequestRate().mark();
```

Comment:
What is the purpose of this being marked here?

Reply:
Oh, I see, every time an async read is done we create a new RemoteLogReader. Still, is it possible to do this increment as part of the call method?

Reply:
This could be another improvement. I  think this is out of the scope of this PR.